<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>advancedImageWidget</class>
 <widget class="QWidget" name="advancedImageWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1089</width>
    <height>812</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>1</number>
   </property>
   <property name="topMargin">
    <number>1</number>
   </property>
   <property name="rightMargin">
    <number>1</number>
   </property>
   <property name="bottomMargin">
    <number>1</number>
   </property>
   <property name="spacing">
    <number>1</number>
   </property>
   <item row="0" column="0">
    <widget class="QSplitter" name="splitter">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QWidget" name="controlsWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>80</height>
       </size>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <property name="spacing">
        <number>1</number>
       </property>
       <item row="1" column="0" colspan="2">
        <widget class="QFrame" name="frame_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="spacing">
           <number>3</number>
          </property>
          <property name="leftMargin">
           <number>3</number>
          </property>
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item>
           <widget class="QLabel" name="sizeNamelabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>17</height>
             </size>
            </property>
            <property name="text">
             <string>Info:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="infoValueLabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>17</height>
             </size>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>---</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="fileNamelabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>17</height>
             </size>
            </property>
            <property name="text">
             <string>File name:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="fileValuelabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>17</height>
             </size>
            </property>
            <property name="text">
             <string>---</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QWidget" name="widget_2" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QToolButton" name="freezeButton">
            <property name="toolTip">
             <string>Freeze</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/prefix1/pause.png</normaloff>:/new/prefix1/pause.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="autoRaise">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="spacing">
              <number>3</number>
             </property>
             <property name="leftMargin">
              <number>3</number>
             </property>
             <property name="topMargin">
              <number>3</number>
             </property>
             <property name="rightMargin">
              <number>3</number>
             </property>
             <property name="bottomMargin">
              <number>3</number>
             </property>
             <item>
              <widget class="QToolButton" name="selectZoomButton">
               <property name="toolTip">
                <string>Select a part to zoom</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../../resources/main.qrc">
                 <normaloff>:/new/prefix1/zoom_layer.png</normaloff>:/new/prefix1/zoom_layer.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>24</height>
                </size>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="autoExclusive">
                <bool>true</bool>
               </property>
               <property name="autoRaise">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="panButton">
               <property name="toolTip">
                <string>Pan image</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../../resources/main.qrc">
                 <normaloff>:/new/prefix1/hand.png</normaloff>:/new/prefix1/hand.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>24</height>
                </size>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
               <property name="autoExclusive">
                <bool>true</bool>
               </property>
               <property name="autoRaise">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="infoButton">
               <property name="toolTip">
                <string>Info</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../../resources/main.qrc">
                 <normaloff>:/new/prefix1/info.png</normaloff>:/new/prefix1/info.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>24</height>
                </size>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="autoExclusive">
                <bool>true</bool>
               </property>
               <property name="autoRaise">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="fitToWindowCheckBox">
            <property name="toolTip">
             <string>Fit To View Area</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/images/layer_resize.png</normaloff>
              <normalon>:/new/images/layer_resize_actual.png</normalon>:/new/images/layer_resize.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="aspectCheckBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Keep Aspect</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/images/zoom_actual_equal.png</normaloff>
              <normalon>:/new/images/zoom_extend.png</normalon>:/new/images/zoom_actual_equal.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="ExponentialSlider" name="expSpinBox" native="true">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="zoomOutButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Zoom Out</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/prefix1/zoom_out.png</normaloff>:/new/prefix1/zoom_out.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="autoRaise">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="zoomInButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Zoom In</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/prefix1/zoom_in.png</normaloff>:/new/prefix1/zoom_in.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="autoRaise">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="resetZoomButton">
            <property name="toolTip">
             <string>ResetZoom</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/prefix1/refresh.png</normaloff>:/new/prefix1/refresh.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="autoRaise">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="rotationComboBox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>Original</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>90°</string>
             </property>
             <property name="icon">
              <iconset resource="../../resources/main.qrc">
               <normaloff>:/new/prefix1/transform_rotate_90.png</normaloff>:/new/prefix1/transform_rotate_90.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>180°</string>
             </property>
             <property name="icon">
              <iconset resource="../../resources/main.qrc">
               <normaloff>:/new/prefix1/transform_rotate_180.png</normaloff>:/new/prefix1/transform_rotate_180.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>270°</string>
             </property>
             <property name="icon">
              <iconset resource="../../resources/main.qrc">
               <normaloff>:/new/prefix1/transform_rotate_270.png</normaloff>:/new/prefix1/transform_rotate_270.png</iconset>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="saveButton">
            <property name="toolTip">
             <string>Save Image</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/prefix1/picture_save.png</normaloff>:/new/prefix1/picture_save.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="autoRaise">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="saveFlowButton">
            <property name="toolTip">
             <string>Save Sequence</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../../resources/main.qrc">
              <normaloff>:/new/prefix1/film_save.png</normaloff>:/new/prefix1/film_save.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="autoRaise">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="ViAreaWidget" name="widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ViAreaWidget</class>
   <extends>QWidget</extends>
   <header>viAreaWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ExponentialSlider</class>
   <extends>QWidget</extends>
   <header>widgets/exponentialSlider.h</header>
   <container>1</container>
   <slots>
    <signal>valueChanged(double)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../resources/main.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>freezeButton</sender>
   <signal>released()</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>freezeImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>100</x>
     <y>78</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>released()</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>saveImageToFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1043</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>748</x>
     <y>8</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveFlowButton</sender>
   <signal>released()</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>stopSaveFlow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1082</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>780</x>
     <y>67</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomInButton</sender>
   <signal>released()</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>345</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>780</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomOutButton</sender>
   <signal>released()</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>780</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetZoomButton</sender>
   <signal>released()</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>zoomReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>379</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>780</x>
     <y>123</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>expSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>zoomChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>277</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>741</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotationComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>zoomChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>603</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aspectCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>fitToggled()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>1087</x>
     <y>114</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fitToWindowCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>advancedImageWidget</receiver>
   <slot>fitToggled()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>171</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>1089</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fitToWindowCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>aspectCheckBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>freezeImage()</slot>
  <slot>saveImageToFile()</slot>
  <slot>stopSaveFlow()</slot>
  <slot>zoomIn()</slot>
  <slot>zoomOut()</slot>
  <slot>fitToggled()</slot>
  <slot>zoomChanged()</slot>
  <slot>zoomReset()</slot>
  <slot>toolButtonReleased()</slot>
 </slots>
</ui>
